### ISSUE TITLE
Expand Lint/Void to catch other void contexts, specifically inside `if` statements

### ISSUE THREAD

**Issue by jez** [rubocop/rubocop#12393]
## Is your feature request related to a problem? Please describe.

**Version**

```
❯ rubocop -V
1.57.2 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.2) [arm64-darwin22]
```

**foo.rb**

```ruby
# frozen_string_literal: true

def example
  ''
  if T.unsafe(true)
    puts
    ''
  end
  nil
end
```

**Output**
```
❯ rubocop foo.rb
Inspecting 1 file
W

Offenses:

foo.rb:4:3: W: [Correctable] Lint/Void: Literal '' used in void context.
  ''
  ^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

In this example, we see that the `''` on the first line of the `example` method reports a `Lint/Void` violation.

## Describe the solution you'd like

The `''` inside the `if` is also useless—it would be nice to have a rule which catches problems like this. This sort of code usually indicates that someone expected to be assigning `''` to a variable, but no assignment is happening.

## Describe alternatives you've considered

One solution would be adding this to `Lint/Void`.
Another solution would be creating a new rule for this.

## Additional context

n/a
